src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProjectWithGeneratePackageOnBuild.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Xml.Linq;
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using NuGet.Packaging;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.ToolPack.Tests
+{
+    public class GivenThatWeWantToPackAToolProjectWithGeneratePackageOnBuild : SdkTest
+    {
+
+        private const string AppName = "consoledemo";
+
+        public GivenThatWeWantToPackAToolProjectWithGeneratePackageOnBuild(ITestOutputHelper log) : base(log)
+        {}
+
+        private TestAsset SetupAndRestoreTestAsset([CallerMemberName] string callingMethod = "")
+        {
+            TestAsset testAsset = _testAssetsManager
+                .CopyTestAsset("PortableToolWithP2P", callingMethod)
+                .WithSource()
+                .WithProjectChanges((projectPath, project) =>
+                {
+                    if (IsAppProject(projectPath))
+                    {
+                        XNamespace ns = project.Root.Name.Namespace;
+                        XElement propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
+                        propertyGroup.Add(new XElement(ns + "GeneratePackageOnBuild", "true"));
+                    }
+                })
+                .Restore(Log, "App");
+
+            return testAsset;
+        }
+
+        [Fact]
+        public void It_builds_successfully()
+        {
+            TestAsset testAsset = SetupAndRestoreTestAsset();
+            var appProjectDirectory = Path.Combine(testAsset.TestRoot, "App");
+            var buildCommand = new BuildCommand(Log, appProjectDirectory);
+
+            CommandResult result = buildCommand.Execute();
+
+            result.StdOut.Should().NotContain("There is a circular dependency");
+            result.ExitCode.Should().Be(0);
```

Comment:
This is written elsewhere as Should().Pass().And.NotHaveStdOutContaining("...");

Reply:
fixed